src/elements/content-preview/ContentPreview.js

```diff
@@ -1046,10 +1048,13 @@ class ContentPreview extends PureComponent<Props, State> {
     /**
      * Handles version change events
      *
-     * @param {string} versionId - The version id to set as current
+     * @param {string} [version] - The version that is now previewed
+     * @param {object} [additionalVersionInfo] - extra info about the version
      */
-    onVersionChange = (versionId?: string) => {
-        this.setState({ selectedVersionId: versionId });
+    onVersionChange = (version?: BoxItemVersion, additionalVersionInfo?: Object): void => {
+        const { onVersionChange }: Props = this.props;
+        onVersionChange(version, additionalVersionInfo);
+        this.setState({ selectedVersionId: getProp(version, 'id') });
```

Comment:
When would `version` be undefined? Is `selectedVersionId` as undefined a valid state?

Reply:
Whenever the `versionSidebarContainer` unmounts, it calls onVersionChange() with no version ID. It also gets set to undefined directly when ContentPreview destroys preview. I think it's used as a way to clear the preview logic around needing to reload, and it can be used to set the header back to the normal one before when a fresh preview is happening. 

in `shouldLoadPreview`,  ``` if (selectedVersionId !== prevSelectedVersionId) {```

will always be true when the prev ID is undefined, which will force preview to reload. The header will also be forced to reset since it has no version information.

Reply:
👍 Thanks for the explanation!